Let users with manage permissions to map/validate

ramyaragupathy opened the pull request:
As reported by Ralph, for projects with team permissions on mapping/validation  - admin users were not permitted to lock tasks for validation.
 
Expected behaviour: users with managerial permissions for a project - admins, org managers, PM team members, author can choose to map/validate tasks irrespective of whether they're part of mapper and validation team.

Permission matrix  for mapping a project between users at various access levels and mapping_permissions set in the backend:

User Access | LEVEL - Only users with intermediate or advanced level | TEAMS - Only team members | TEAMS_LEVEL - Only intermediate and advanced team members
-------------|---------------------------------------------------------|-------------------------------|-------------------------------------------------------------------
ADMIN| ✅ |✅|✅
PROJECT AUTHOR|✅|✅|✅
PROJECT ORG MANAGER|✅|✅|✅
MAPPER TEAM|N/A|✅|Only if user level matches
VALIDATOR TEAM|N/A|✅|Only if user level matches
PM TEAM|✅|✅|✅
TM Mappers outside the team|Only if user level matches| ❌ | ❌ 

Permission matrix  for validating a project between users at various access levels and validation_permissions set in the backend:

User Access | LEVEL - Only users with intermediate or advanced level | TEAMS - Only team members | TEAMS_LEVEL - Only intermediate and advanced team members
-------------|---------------------------------------------------------|-------------------------------|-------------------------------------------------------------------
ADMIN| ✅ |✅|✅
PROJECT AUTHOR|✅|✅|✅
PROJECT ORG MANAGER|✅|✅|✅
MAPPER TEAM|N/A| ❌ | ❌ 
VALIDATOR TEAM|Only if user level matches|✅|Only if user level matches
PM TEAM|✅|✅|✅
General TM Mappers|Only if user level matches| ❌ | ❌

willemarcel commented:
@ramyaragupathy I updated the permission verification on frontend and added the organisation field on the project summary endpoint. Could you test it?

ramyaragupathy commented:
> @ramyaragupathy I updated the permission verification on frontend and added the organisation field on the project summary endpoint. Could you test it?

Tested, works good

russdeffner commented:
After initial launch, we have found that Private projects were maybe missed in this structure.
